pacehrh/R/pace_cadre_process.R

```diff
@@ -0,0 +1,282 @@
+    monthly <- as.vector(matrix(data = curve, ncol = 1) %*% matrix(data = annual, nrow = 1))
+    monthlyValuesMatrix[i,] <- monthly
+  }
+
+  monthStrs <- as.character(1:12)
+  s <- sapply(as.character(BVE$years), function(year) {
+    paste0(year, ".", monthStrs)
+  })
+
+  dimnames(monthlyValuesMatrix) <- list("Role" = roles$RoleID, "Month" = as.vector(s))
+
+  return(list(annualOverheads = annualValuesMatrix, monthlyOverheads = monthlyValuesMatrix))
+}
+
+#' Compute And Save Suite Results In Extended Format
+#'
+#' @param results Results structure (as returned by [RunExperiments()]).
+#' @param filepath Destination CSV file to write results. Default = NULL.
+#' @param run Name of experiment run
+#'
+#' @return Data frame version of CSV file.
+#' @export
+#'
+#' @md
+#' @examples
+#' \dontrun{
+#' }
+
+SaveExtendedSuiteResults <- function(results = NULL, filepath = NULL, run = "Run-1") {
+  if (is.null(results)) {
+    return(NULL)
+  }
+
+  trialIds <- names(results)
+  scenario <- BVE$scenario$UniqueID
+
+  # Create the usual output CSV file format
+  l <- lapply(seq_along(trialIds), function(index) {
+    r <- results[[index]]
+    if ("SeasonalityResults" %in% names(r)) {
+      return(.saveSuiteResults2(r, scenario, index, run))
+    } else {
+      return(.saveSuiteResults1(r, scenario, index, run))
+    }
+  })
+
+  out <- data.table::rbindlist(l)
+
+  # Convert Service_time from minutes to hours
+  out$Service_time <- out$Service_time / 60.0
+
+  # Grab the Tasks table
+  tasks <- BVE$taskData
+  data.table::setDT(tasks)
+
+  # Add a total number-of-contacts value to the Tasks table
+  tasks$NumContactsAnnual[is.na(tasks$NumContactsAnnual)] <- 0
+  tasks$NumContactsPerUnit[is.na(tasks$NumContactsPerUnit)] <- 0
+  tasks$NumContactsPer <-
+    tasks$NumContactsPerUnit + tasks$NumContactsAnnual
+
+  # Filter the Tasks table column set
+  tasks <-
+    tasks[, .(Indicator,
+              CommonName,
+              ClinicalOrNon,
+              ClinicalCat,
+              ServiceCat,
+              NumContactsPer)]
+
+  # Join Scenario information to the output
+  out <- dplyr::left_join(out,
+                          BVE$scenario[c("UniqueID",
+                                         "WeeksPerYr",
+                                         "HrsPerWeek",
+                                         "BaselinePop",
+                                         "DeliveryModel")],
+                          by = dplyr::join_by(Scenario_ID == UniqueID))
+
+  # Join Task information to the output
+  out <- dplyr::left_join(out,
+                          tasks,
+                          by = dplyr::join_by(Task_ID == Indicator))
+
+  # Create a stochastic hours per week value that's different for each year.
+  sdvalue <-
+    BVE$stochasticParams$p[BVE$stochasticParams$Value == "Hours per week"]
+
+  # TODO: find a more efficient calculation.
+  stochasticHrs <- out[, .(Count = .N), by = .(Trial_num, Year)]
+
+  stochasticHrs$HPW_stochastic <-
```

Comment:
We need documentation to explain why do we need to calculate this here, also from design perspective,  it seems odd to do calculation in "save results" methods.

Reply:
This was in the original post-processing, so I kept it. But I agree, it doesn't make a lot of sense here (or anywhere else for that matter.)

Reply:
@celiot-IDM  We cannot delete this, at least not as part of this pull request. We may eventually make the design decision that we want to remove the hours per week and weeks per year entirely, but that is a larger choice that involves removing scenarios columns etc. as well. I'm not convinced we want to do that at this stage. @MeWu-IDM if we want to move this elsewhere, please feel free to make that change as you see fit.

Reply:
Thank you @BHagedorn-IDM .

The logic for keeping the HPW_stochastic computation goes like this: the SaveExtendedSuiteResults() function is a "widening" function. In other words, it adds columns to the standard suite results. But this reasoning, HPW_stochastic is just another column.

The counter-argument is that we already have a place in the main code that does all the stochastic calculations, so why sneak another stochastic calculation in at the last moment. But this reasoning, the HPW_stochastic computation should be moved to be part of the RunExperiment() processing. Unfortunately, that's a bigger lift than just tacking it on at the end. So if we can live with this sub-optimal implementation for now, we should.